src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Formatters/EnumAsStringFormatter`1.cs

```diff
@@ -52,4 +52,67 @@ public T Deserialize(ref MessagePackReader reader, MessagePackSerializerOptions
             return value;
         }
     }
+
+    public sealed class GenericEnumFormatter<T> : IMessagePackFormatter<T>
+        where T : Enum
+    {
+        delegate void EnumSerialize(ref MessagePackWriter writer, ref T value);
+        delegate T EnumDeserialize(ref MessagePackReader reader);
+
+        readonly EnumSerialize serializer;
+        readonly EnumDeserialize deserializer;
+
+        public GenericEnumFormatter()
```

Comment:
Invoking delegates tends to be slower than most other method call types. I think they're slower than virtual/interface methods.
If instead of exposing a public ctor, if you instead expose a public static factory method, you'd have the freedom later to refactor this to return a private derived type based on `T`'s size in the future without an API breaking change.

Reply:
This is support for registering "manually" for AOT, IL2CPP.
(`new GenericEnumFormatter <T>` is easier than mpc).
For example, MagicOnion generates the same Formatter as mpc,
https://github.com/Cysharp/MagicOnion/blob/master/src/MagicOnion.UniversalCodeGenerator/Generator/EnumTemplate.tt
I wanted to register this with new ...

However, the first reason is to pass the test quickly.